spec/system/claims/schools/mentors/claims_school_user_adds_mentor_to_school_spec.rb

```diff
@@ -0,0 +1,267 @@
+require "rails_helper"
+
+RSpec.describe "Claims school user adds mentors to schools", service: :claims, type: :system do
+  scenario do
+    given_an_eligible_school_exists_with_an_assigned_mentor_and_an_unassigned_one
+    and_given_the_get_teacher_api_is_stubbed
+    and_i_am_signed_in
+
+    when_i_navigate_to_mentors
+    then_i_see_the_mentors_index_page
+
+    when_i_click_on_add_mentor
+    then_i_see_the_add_mentor_page
+
+    when_i_click_on_back
+    then_i_see_the_mentors_index_page
+
+    when_i_click_on_add_mentor
+    then_i_see_the_add_mentor_page
+
+    when_i_click_cancel
+    then_i_see_the_mentors_index_page
+
+    when_i_click_on_add_mentor
+    then_i_see_the_add_mentor_page
+    and_i_click_on_the_help_text
+    then_i_see_link_to_trn_guidance
+
+    when_i_navigate_to_mentors
+    and_i_click_on_add_mentor
+    and_i_click_on_continue
+    then_i_see_a_validation_error_that_i_must_add_a_trn_and_date_of_birth
+
+    when_i_enter_an_invalid_trn
+    and_i_click_on_continue
+    then_i_see_a_validation_error_that_i_must_enter_a_seven_digit_trn
+
+    when_i_enter_the_trn_and_dob_of_an_existing_mentor_at_this_school
+    and_i_click_on_continue
+    then_i_see_a_validation_error_that_the_mentor_is_already_added
+
+    when_i_enter_the_trn_of_an_existing_mentor_not_at_this_school
+    and_i_enter_the_wrong_date_of_birth
+    and_i_click_on_continue
+
+    then_i_see_the_no_results_page
+
+    when_i_navigate_to_mentors
+    and_i_click_on_add_mentor
```

Comment:
```suggestion
    when_i_navigate_to_mentors
    and_i_click_on_add_mentor
    then_i_see_the_add_mentor_page
```

We should be checking that we're on the right page after clicking the button here